controllers/statussync/policy_status_sync.go

```diff
@@ -441,7 +490,230 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
+	apiURL string,
+	events workqueue.RateLimitingInterface,
+) error {
+	managedClient, err := dynamic.NewForConfig(managedCfg)
+	if err != nil {
+		return err
+	}
+
+	var clusterID string
+
+	idClusterClaim, err := managedClient.Resource(clusterClaimGVR).Get(ctx, "id.k8s.io", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	if err == nil {
+		clusterID, _, _ = unstructured.NestedString(idClusterClaim.Object, "spec", "value")
+	}
+
+	if clusterID == "" {
+		log.Info("The id.k8s.io cluster claim is not set. Using the cluster ID of unknown.")
+
+		clusterID = "unknown"
+	}
+
+	processedEvents := map[types.UID]bool{}
+
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(hubCfg.CAData)
+
+	httpClient := &http.Client{Timeout: 60 * time.Second}
+
+	if hubCfg.CertData != nil && hubCfg.KeyData != nil {
+		log.Info("Using certificate authentication with the compliance API server")
+
+		cert, err := tls.X509KeyPair(hubCfg.CertData, hubCfg.KeyData)
+		if err != nil {
+			log.Error(err, "Failed to load the hub kubeconfig for certificate authentication on the compliance API")
+
+			return err
+		}
+
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				MinVersion: tls.VersionTLS12,
+				Certificates: []tls.Certificate{
+					{
+						Certificate: cert.Certificate,
+						PrivateKey:  cert.PrivateKey,
+					},
+				},
+				RootCAs: caCertPool,
+			},
+		}
+	} else {
+		httpClient.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				MinVersion: tls.VersionTLS12,
+				RootCAs:    caCertPool,
+			},
+		}
+	}
+
+	for {
+		ceUntyped, shutdown := events.Get()
+		if shutdown {
+			return nil
+		}
+
+		ce, ok := ceUntyped.(utils.ComplianceAPIEventRequest)
+		if !ok {
+			events.Forget(ceUntyped)
+			events.Done(ceUntyped)
+
+			continue
+		}
+
+		if ce.UID != "" && processedEvents[ce.UID] {
+			events.Forget(ceUntyped)
+			events.Done(ceUntyped)
+
+			continue
+		}
+
+		ce.Cluster = utils.ComplianceAPIEventCluster{
+			Name:      clusterName,
+			ClusterID: clusterID,
+		}
+
+		requestJSON, err := json.Marshal(ce)
+		if err != nil {
+			log.Error(err, "Failed to record the event with the compliance API due to an invalid format")
+
+			events.Forget(ceUntyped)
+			events.Done(ceUntyped)
+
+			continue
+		}
+
+		httpRequest, err := http.NewRequestWithContext(
+			ctx, http.MethodPost, apiURL+"/api/v1/compliance-events", bytes.NewBuffer(requestJSON),
+		)
+		if err != nil {
+			log.Error(err, "Failed to record the event with the compliance API")
+
+			events.AddRateLimited(ceUntyped)
+			events.Done(ceUntyped)
+
+			continue
+		}
+
+		httpRequest.Header.Set("Content-Type", "application/json")
+
+		if hubCfg.BearerToken != "" {
+			httpRequest.Header.Set("Authorization", "Bearer "+hubCfg.BearerToken)
+		}
+
+		httpResponse, err := httpClient.Do(httpRequest)
+		if err != nil {
+			log.Info("Failed to record the compliance event with the compliance API. Will requeue.")
```

Comment:
Need check status check? if it is 500 do not add to `events.AddRateLimited`

Reply:
An error here indicates a connection issue and not an error status code. In this case we want to retry. In the case of unexpected status codes, we want to continue retrying. If it's unauthorized, it's a misconfiguration that needs to be fixed and once fixed, we'll be able to send the events.